### PR TITLE
[12.0][IMP] hr_timesheet_sheet: hooks

### PR DESCRIPTION
**Pull request by alexey-pelykh**
@MiquelRForgeFlow @astirpe this is a slight change to allow extension modules to work with default period start/end dates more elegantly + would allow other modules to query "what period would it be for this date" w/o sheet existing.

### CODE REVIEW

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py [hunk 1]
```diff
@@ -28,35 +28,16 @@ class Sheet(models.Model):
...
-                delta = relativedelta(weekday=(int(
-                    user.company_id.timesheet_week_start) + 6) % 7)
-            else:
-                delta = relativedelta(days=6-today.weekday())
-            return today + delta
-        elif r == MONTHLY:
-            return today + relativedelta(months=1, day=1, days=-1)
-        return today
+        return self._get_period_end(
+            self.env['res.users'].browse(self.env.uid).company_id,
```
Comment (astirpe) [inline hr_timesheet_sheet/models/hr_timesheet_sheet.py]:
```suggestion
            self.env.user.company_id,
```

**Review by astirpe — CHANGES_REQUESTED** [review]

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM, after applying @astirpe suggestion

**Comment by alexey-pelykh** [comment]
@MiquelRForgeFlow @astirpe done :) I guess this should be minor version bump?

**Comment by MiquelRForgeFlow** [comment]
yes

**Pull request merged**